pyquil/api/_quantum_computer.py

```diff
@@ -273,6 +272,8 @@ def compile(self, program: Program,
             to protoquil (executable on QPU). A value of ``None`` means defer to server.
         :return: An executable binary suitable for passing to :py:func:`QuantumComputer.run`.
         """
+        self.reset()
```

Comment:
I'd like you to update the documentation of this function, since the inclusion of `.reset()` is nonobvious.

It feels to me like the inclusion of `.reset()` here is a crutch to avoid rewriting a bunch of examples where we might have otherwise explicitly expected a user to request a reset. My preference for the perspective of Pristine Code is to update the examples, but from a practical standpoint, I certainly see the argument for making this change instead.

Reply:
Your assumption is correct -- by slipping a reset into compile we can cover a lot of potential issues, as users must always compile before running on the QPU. It is certainly a little sneaky, but I don't really see how it could affect the user in a negative way?

Reply:
It won't affect the current QPU user in a negative way because the QPU has such a small lifetime. What about a QVM user running against a persistent (perhaps local) QVM?

Reply:
Hmm yeah I see your point. We could certainly drop it for now and open an issue to either improve documentation for `qc.reset()` or think or some clever way to ensure this doesn't impact performance?

Reply:
Ok I think I found a "best of both worlds" solution -- check it out.